### PR TITLE
Fix tachyon across z-levels

### LINKED ISSUE #58380: tachyon-doppler arrays do not connect to RD servers on other Z levels
## Round ID:
N/A

## Testmerges:
N/A

## Reproduction:
Load up tramstation
Go to your launch site
Check out the tachyon
image
Get rekt out of doing toxins experiments

### PR DESCRIPTION
**Pull request by RandomGamer123**
## About The Pull Request
Tachyon-doppler arrays now work when on a different z-level than the RD server provided both of them are on a station z.
Same applies for all things using the experiment handler component in fact.

## Why It's Good For The Game
Fixes #58380

## Changelog
:cl:
fix: Tachyon-doppler arrays now connect with servers on different z-levels.
/:cl:

### CODE REVIEW

File: code/modules/experisci/experiment/handlers/experiment_handler.dm [hunk 1]
```diff
@@ -298,7 +298,7 @@
 	var/list/local_servers = list()
 	for (var/obj/machinery/rnd/server/server in SSresearch.servers)
 		var/turf/position_of_this_server_machine = get_turf(server)
-		if (turf_to_check_for_servers && position_of_this_server_machine && position_of_this_server_machine.z == turf_to_check_for_servers.z)
+		if (turf_to_check_for_servers && position_of_this_server_machine && SSmapping.level_trait(position_of_this_server_machine.z,ZTRAIT_STATION) && SSmapping.level_trait(turf_to_check_for_servers.z,ZTRAIT_STATION))
```
Comment (LemonInTheDark) [inline code/modules/experisci/experiment/handlers/experiment_handler.dm:301]: You should use early returns here so you can split up this check to keep line length down

**Comment by LemonInTheDark** [comment]
> Same applies for all things using the experiment handler component in fact.

Are you sure about this? it looks like this just applies to the tachyon

**Comment by RandomGamer123** [comment]
> > Same applies for all things using the experiment handler component in fact.
> 
> Are you sure about this? it looks like this just applies to the tachyon

Yep, much less known version of the bug: Experi-scanners don't work on the lower z-level, but why would anybody do that normally anyways

File: code/modules/experisci/experiment/handlers/experiment_handler.dm [hunk 2]
```diff
@@ -287,18 +287,20 @@
...
  *
  * Arguments:
- * * turf_to_check_for_servers - The turf to get servers on the same z-level of
+ * * turf_to_check_for_servers - The turf to check if its on the station z-level
  */
 /datum/component/experiment_handler/proc/get_available_servers(turf/turf_to_check_for_servers = null)
 	if (!turf_to_check_for_servers)
 		turf_to_check_for_servers = get_turf(parent)
 	var/list/local_servers = list()
+	if (SSmapping.level_trait(turf_to_check_for_servers.z,ZTRAIT_STATION))
```
Comment (AnturK) [inline code/modules/experisci/experiment/handlers/experiment_handler.dm:299]: Did you drop a negation here ? I assume it's supposed to skip servers off-station ?

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/experisci/experiment/handlers/experiment_handler.dm (hunk 2 above)
Comment (RandomGamer123) [reply code/modules/experisci/experiment/handlers/experiment_handler.dm]: Yep

**Pull request merged**